index.js

```diff
@@ -144,132 +127,158 @@ class Disk {
 	}
 	
 	discard(offset, length, callback) {
-		console.log('UNIMPLEMENTED: discarding', length, 'bytes at offset', offset);  // eslint-disable-line no-console
+		this._insertDiskChunk(new diskchunk.DiscardDiskChunk(offset, length));
 		callback(null);
 	}
 
 	getCapacity(callback) {
-		this._getCapacity(callback);
+		if (this.capacity !== null) {
+			callback(null, this.capacity);
+			return;
+		}
+		const self = this;
+		this._getCapacity(function(err, capacity) {
+			if (err) {
+				callback(err);
+				return;
+			}
+			self.capacity = capacity;
+			callback(null, capacity);
+		});
 	}
 
-	getStream(highWaterMark, callback) {
-		if ((typeof highWaterMark === 'function') && (callback === undefined)) {
-			callback = highWaterMark;
+	getStream(...argv) {
+		// args: ([position, [length, [highWaterMark]]], callback)
+		const callback = argv.pop();
+		let [ position, length, highWaterMark ] = argv;
+		position = Number.isInteger(position) ? position : 0;
+		if (!Number.isInteger(highWaterMark)) {
 			highWaterMark = DEFAULT_HIGH_WATER_MARK;
 		}
 		const self = this;
-		self.getCapacity(function(err, capacity) {
+		self.getCapacity(function(err, end) {
 			if (err) {
 				callback(err);
 				return;
 			}
-			callback(null, new DiskStream(self, capacity, highWaterMark));
+			if (Number.isInteger(length)) {
+				end = Math.min(position + length, end);
+			}
+			callback(null,new DiskStream(self, end, highWaterMark, position));
 		});
 	}
 
-	_insertDiskChunk(buffer, offset) {
-		const write = new DiskChunk(buffer, offset);
-		if (this.knownChunks.length === 0) {
-			// Special case for empty list: insert and return.
-			this.knownChunks.push(write);
-			return;
-		}
-		let firstFound = false;
-		let other;
-		for (let i = 0; i < this.knownChunks.length; i++) {
+	getDiscardedChunks() {
+		return this.knownChunks.filter(function(chunk) {
+			return (chunk instanceof diskchunk.DiscardDiskChunk);
+		});
+	}
+
+	getBlockMap(blockSize, calculateChecksums, callback) {
+		const self = this;
```

Comment:
NIT: same here

Reply:
This will be fixed in the forthcoming TypeScript PR.